src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java

```diff
@@ -0,0 +1,73 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl.checks;
+
+
+import com.cognifide.aemrules.metadata.Metadata;
+import com.cognifide.aemrules.tag.Tags;
+import com.cognifide.aemrules.version.AemVersion;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.scripting.sightly.compiler.expression.Expression;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(
+    key = CamelCaseCheck.RULE_KEY,
+    name = CamelCaseCheck.RULE_MESSAGE,
+    priority = Priority.MINOR,
+    tags = Tags.AEM
+)
+@AemVersion(
+    from = "6.0"
+)
+@Metadata(
+    technicalDebt = "5min"
+)
+public class CamelCaseCheck extends AbstractHtlCheck {
+
+    public static final String RULE_KEY = "HTL-3";
+
+    static final String RULE_MESSAGE = "Use Camel Case in identifiers";
+
+    private static final String CAMEL_CASE_PATTERN = "([a-z]+[A-Z]+\\w+)+";
```

Comment:
If you don't need to capture the group () then use ?: ( it means that the group is matched but is not captured for back-referencing)  (?:[a-z]+[A-Z]+\\w+)+